_pytadbit/parsers/cooler_parser.py

```diff
@@ -0,0 +1,410 @@
+"""
+September 2, 2019.
+
+"""
+
+from __future__ import division
+import os
+import datetime
+import json
+import h5py
+import numpy as np
+from time import time
+
+def printime(msg):
+    print (msg +
+           (' ' * (79 - len(msg.replace('\n', '')))) +
+           '[' +
+           str(datetime.datetime.fromtimestamp(time()).strftime('%Y-%m-%d %H:%M:%S')) +
+           ']')
+
+class cooler_file(object):
+    """
+        Cooler file wrapper.
+    """
+    def __init__(self, outcool, resolution, sections, regions,
+                 h5opts=None, verbose=False):
+        """
+        :param outcool: path to cool file to be created
+        :param resolution: resolution of the matrix
+        :param sections: dictionary with chromosomes and lengths
+        :param regions: chromosomes present in the matrix
+        :param None h5opts: options for the h5py outcool file
+        :param True verbose: speak
+
+        """
+
+        self.root_grp = 'resolutions'
+        self.resolution = resolution
+        if not os.path.exists(outcool):
+            with h5py.File(outcool, "w") as f:
+                grp = f.create_group(self.root_grp)
+                grp.create_group(str(self.resolution))
+            if verbose:
+                printime('Opening cooler %s'%outcool)
+        else:
+            with h5py.File(outcool, "r") as f:
```

Comment:
Did you mean to use `"a"` or `"r+"` mode here? `"r"` mode doesn't allow new group creation.

Reply:
Yes, you are right. For the moment the code just exports the matrix at the given resolution (just one resolution) and we open the cooler file only to include the weights.
That's why I didn't see the error.